api/handler/delete.go

```diff
@@ -213,7 +213,10 @@ func (h *handler) DeleteMultipleObjectsHandler(w http.ResponseWriter, r *http.Re
 
 	var errs []error
 	for _, obj := range deletedObjects {
-		if obj.Error == nil {
+		if obj.Error == nil || (len(obj.VersionID) == 0 && errors.IsS3Error(obj.Error, errors.ErrInvalidVersion)) {
+			if requested.Quiet {
+				continue
+			}
```

Comment:
If quiet mode was chosen in a request we don't add successfull object deletion to our response.